pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -112,3 +132,193 @@ func SetDefaults_LeaderElectionConfiguration(obj *LeaderElectionConfiguration) {
 		obj.RetryPeriod = unversioned.Duration{Duration: 2 * time.Second}
 	}
 }
+
+func SetDefaults_KubeletConfiguration(obj *KubeletConfiguration) {
+	if obj.Address == "" {
+		obj.Address = "0.0.0.0"
+	}
+	if obj.CAdvisorPort == 0 {
+		obj.CAdvisorPort = 4194
+	}
+	if obj.VolumeStatsAggPeriod == zeroDuration {
+		obj.VolumeStatsAggPeriod = unversioned.Duration{Duration: time.Minute}
+	}
+	if obj.CertDirectory == "" {
+		obj.CertDirectory = "/var/run/kubernetes"
+	}
+	if obj.ConfigureCBR0 == nil {
+		obj.ConfigureCBR0 = boolVar(false)
+	}
+	if obj.ContainerRuntime == "" {
+		obj.ContainerRuntime = "docker"
+	}
+	if obj.RuntimeRequestTimeout == zeroDuration {
+		obj.RuntimeRequestTimeout = unversioned.Duration{Duration: 2 * time.Minute}
+	}
+	if obj.CPUCFSQuota == nil {
+		obj.CPUCFSQuota = boolVar(true)
+	}
+	if obj.DockerExecHandlerName == "" {
+		obj.DockerExecHandlerName = "native"
+	}
+	if obj.DockerEndpoint == "" { // TODO: This isn't in NewKubeletServer in options.go in master...
+		obj.DockerEndpoint = "unix:///var/run/docker.sock"
```

Comment:
"" is not valid. Defaulting it to "unix:///var/run/docker.sock" when you get a "" SGTM. Also DockerAPIEndpoint sounds fine. That struct was very literally translated from the cli options so you should change it to be more consistent.